### PR TITLE
Improve error messages for StackReference Get*Output

### PR DESCRIPTION
**Pull request by lukehoban**
Add more context to the error messages returned from `Get*Output` methods on `StackReference`.

Also quotes the property name in the "awaiting input property" RPC message.

Part of https://github.com/pulumi/pulumi/issues/10476.

### CODE REVIEW

File: sdk/go/pulumi/rpc.go [hunk 1]
```diff
@@ -67,22 +67,22 @@ func mapStructTypes(from, to reflect.Type) func(reflect.Value, int) (reflect.Str
...
-//			  Cust2   Cust3  Comp3
-//			  /                 \
-//		  Cust4                Cust5
+//			  Comp1
+//		  /     |     \
+//	  Cust1   Comp2  Remote1
+//			  /   \       \
+//		  Cust2   Cust3  Comp3
+//		  /                 \
+//	  Cust4                Cust5
```
Comment (lukehoban) [inline sdk/go/pulumi/rpc.go:85]: These changes were triggered by my IDE.  I can undo them if they are not something that new Go toolchain is forcing on anyone who touches this file?

File: sdk/go/pulumi/stack_reference.go [hunk 2]
```diff
@@ -53,11 +53,16 @@ func (s *StackReference) GetOutput(name StringInput) AnyOutput {
 
 // GetStringOutput returns a stack output keyed by the given name as an StringOutput
 func (s *StackReference) GetStringOutput(name StringInput) StringOutput {
-	return s.GetOutput(name).ApplyT(func(out interface{}) (string, error) {
-		if s, ok := out.(string); ok {
-			return s, nil
+	return All(name, s.GetOutput(name)).ApplyT(func(args []interface{}) (string, error) {
+		name, out := args[0].(string), args[1]
+		if out == nil {
+			return "", fmt.Errorf("stack reference output %q does not exist on stack %q", name, s.name)
```
Comment (lukehoban) [inline sdk/go/pulumi/stack_reference.go:59]: It's a little weird that `GetOutput` doesn't error on missing keys (which seems intentional from comments), but that these helpers *do* error on missing keys.  It seems we made them do so on purpose in https://github.com/pulumi/pulumi/pull/9855.  So I've left that behaviour the same, and just improved the error messages for now.

**Review by iwahbe — APPROVED** [review]
There are some long line lint errors, but beyond that this is great. Once tests are 🟢, LGTM.

**Comment by dixler** [comment]
I can fix the linting errors and update the changelog @lukehoban

**Comment by lukehoban** [comment]
> I can fix the linting errors and update the changelog @lukehoban

Thanks @dixler - but actually missed this before pushing up a change to address these!

**Pull request merged**